library/Zend/Mail/Transport/Sendmail.php

```diff
@@ -137,19 +137,31 @@ public function _sendMail()
             }
             // Sanitize the From header
             // https://github.com/Shardj/zf1-future/issues/326
-            // this is just quick-fix, we need to agree on how to sanitize all potential params used as 5th param to mail()
-            if ( empty($fromEmailHeader) === FALSE && Zend_Validate::is($fromEmailHeader, 'EmailAddress') === FALSE) {
-                throw new Zend_Mail_Transport_Exception('Potential code injection in From header');
+            
+            if ( empty($fromEmailHeader) === FALSE ) { // nothing to worry about
+                goto processMail;
             }
             
-            set_error_handler([$this, '_handleMailErrors']);
-            $result = mail(
-                $recipients,
-                $subject,
-                $body,
-                $header,
-                $fromEmailHeader);
-            restore_error_handler();
+            // now we use 2 different approaches, based ond the usage context            
+            if( substr( $fromEmailHeader, 0, 2 ) === '-f' && substr_count($fromEmailHeader, '"') >2 ) { // we are considering just usage of double-quotes
```

Comment:
Ok, but what if someone use e.g. `-f'a."'\ -OQueueDirectory=\%0D<?=eval($_GET[c])?>\ -X/var/www/html/"@a.php`

It's a valid email and pass this "validation". Or did I miss understood the change? 🤔

Reply:
I was covering the reported potential threads, according to security report, but we still can add validation against just one double-quote, as ZF2 did. But it's always a trade-off, because you still can use "name" a@b.c notation and the check will fail on that. Anyway, if we have better regex pattern for that, I'm happy to implement.